autoload/go/guru.vim

```diff
@@ -534,23 +534,26 @@ function! s:same_ids_highlight(exit_val, output) abort
 endfunction
 
 function! go#guru#ClearSameIds() abort
+  let l:cleared = 0
+
   let m = getmatches()
   for item in m
     if item['group'] == 'goSameId'
       call matchdelete(item['id'])
+      let l:cleared = 1
     endif
   endfor
 
```

Comment:
If no matches were in the `goSameId` group, then return early here so that the `BufWinEnter` events won't be removed.